service/testutil.go

```diff
@@ -0,0 +1,232 @@
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+
+	"github.com/tidepool-org/platform/Godeps/_workspace/src/github.com/ant0ine/go-json-rest/rest"
+)
+
+func panicOnError(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+// MakeSimpleRequest returns a http.Request. The returned request object can be
+// further prepared by adding headers and query string parmaters, for instance.
+func MakeSimpleRequest(method string, urlStr string, body io.Reader) *http.Request {
+
+	r, err := http.NewRequest(method, urlStr, body)
+	panicOnError(err)
+
+	r.Header.Set("Accept-Encoding", "gzip")
+	if body != nil {
+		r.Header.Set("Content-Type", "application/json")
+	}
+
+	return r
+}
+
+// MakeBlobRequest returns a http.Request. The returned request object can be
+// further prepared by adding headers and query string parmaters, for instance.
+func MakeBlobRequest(method string, urlStr string, filename string) *http.Request {
+
+	if filename == "" {
+		r, err := http.NewRequest(method, urlStr, nil)
+		panicOnError(err)
+		return r
+	}
+
+	bodyBuf := &bytes.Buffer{}
+	bodyWriter := multipart.NewWriter(bodyBuf)
+
+	fileWriter, err := bodyWriter.CreateFormFile("uploadfile", filename)
+	panicOnError(err)
+
+	fileHandler, err := os.Open(filename)
+	panicOnError(err)
+
+	_, err = io.Copy(fileWriter, fileHandler)
+	panicOnError(err)
+
+	r, err := http.NewRequest(method, urlStr, bodyBuf)
+
+	panicOnError(err)
+
+	r.Header.Set("Content-Type", bodyWriter.FormDataContentType())
+	r.Header.Set("Accept-Encoding", "gzip")
+
+	bodyWriter.Close()
+
+	return r
+}
+
+// CodeIs compares the rescorded status code
+func CodeIs(r *httptest.ResponseRecorder, expectedCode int) bool {
+	return r.Code == expectedCode
+}
+
+// HeaderIs tests the first value for the given headerKey
+func HeaderIs(r *httptest.ResponseRecorder, headerKey, expectedValue string) bool {
+	value := r.HeaderMap.Get(headerKey)
+	return value == expectedValue
+}
+
+// ContentTypeIsJSON tests that application/json is set
+func ContentTypeIsJSON(r *httptest.ResponseRecorder) bool {
+	return HeaderIs(r, "Content-Type", "application/json")
+}
+
+// ContentEncodingIsGzip tests that gzip is set
+func ContentEncodingIsGzip(r *httptest.ResponseRecorder) bool {
+	return HeaderIs(r, "Content-Encoding", "gzip")
+}
+
+// BodyIs compares the rescorded body
+func BodyIs(r *httptest.ResponseRecorder, expectedBody string) bool {
+	body := r.Body.String()
+	return strings.Trim(body, "\"") == expectedBody
+}
+
+// BodyContains compares the rescorded body
+func BodyContains(r *httptest.ResponseRecorder, expectedToContain string) bool {
+	return strings.Contains(r.Body.String(), expectedToContain)
+}
+
+// DecodeJSONPayload decodes the recorded payload to JSON
+func DecodeJSONPayload(r *httptest.ResponseRecorder, v interface{}) error {
+	content, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(content, v)
```

Comment:
Might want to unmarshal into a local variable and then assign to v at end after checking for no error.